app/src/main/java/org/mozilla/focus/settings/permissions/permissionoptions/SitePermissionOptionsFragment.kt

```diff
@@ -0,0 +1,142 @@
+    private lateinit var defaultSitePermissionOptionsScreenInteractor: DefaultSitePermissionOptionsScreenInteractor
+    private lateinit var hardwarePermissionCheckFeature: HardwarePermissionCheckFeature
+    private lateinit var sitePermissionOptionsStorage: SitePermissionOptionsStorage
+
+    private val sitePermission: SitePermission
+        get() = requireArguments().getParcelable(SITE_PERMISSION)
+            ?: throw IllegalAccessError("Site permission is not set for fragment")
+
+    companion object {
+        const val FRAGMENT_TAG = "SitePermissionOptionsFragment"
+        private const val SITE_PERMISSION = "sitePermission"
+
+        fun addSitePermission(sitePermission: SitePermission): SitePermissionOptionsFragment {
+            val fragment = SitePermissionOptionsFragment()
+            fragment.arguments = Bundle().apply {
+                putParcelable(SITE_PERMISSION, sitePermission)
+            }
+            return fragment
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        sitePermissionOptionsStorage = SitePermissionOptionsStorage(context = requireContext())
+        sitePermissionOptionsScreenStore = SitePermissionOptionsScreenStore(
+            SitePermissionOptionsScreenState(),
+            listOf(
+                SitePermissionOptionsStorageMiddleware(
+                    sitePermission = sitePermission,
+                    storage = sitePermissionOptionsStorage
+                )
+            )
+        )
+        defaultSitePermissionOptionsScreenInteractor = DefaultSitePermissionOptionsScreenInteractor(
+            sitePermissionOptionsScreenStore = sitePermissionOptionsScreenStore
+        )
+        hardwarePermissionCheckFeature = HardwarePermissionCheckFeature(
+            storage = sitePermissionOptionsStorage,
+            store = sitePermissionOptionsScreenStore,
+            sitePermission = sitePermission
+        )
+        lifecycle.addObserver(hardwarePermissionCheckFeature)
+    }
+
+    override val titleText: String
+        get() = sitePermissionOptionsScreenStore.state.sitePermissionLabel ?: ""
+
+    override fun onNavigateUp(): () -> Unit {
+        return {
+            requireComponents.appStore.dispatch(
+                AppAction.OpenSettings(Screen.Settings.Page.SitePermissions)
+            )
+        }
+    }
+
+    @Composable
+    override fun Content() {
+        val sitePermissionOptionsList = sitePermissionOptionsScreenStore.observeAsComposableState { state ->
+            state.sitePermissionOptionList
+        }.value
+        val sitePermissionOptionSelected = sitePermissionOptionsScreenStore.observeAsComposableState { state ->
+            state.selectedSitePermissionOption
+        }.value
+        val isAndroidPermissionGranted = sitePermissionOptionsScreenStore.observeAsComposableState { state ->
+            state.isAndroidPermissionGranted
+        }.value
+        if (
```

Comment:
I don't think it's a good idea . What default value should I put for  val selectedSitePermissionOption: SitePermissionOption

Reply:
That one which should be selected by default at fresh install

Reply:
For this I need to call method permissionSelectedOption() from SitePermissionOptionsStorage that contains a lot of ifs and a reading from SharePref only to put there a non nullable value that will change immediately after that . Besides that  this could also lead to a lot of bugs .

Reply:
What's the default value if you don't have a value in prefs?
" only to put there a non nullable value that will change immediately after that" this is normal behaviour for MVI